### PR TITLE
Refactor LambdaProxyService

### PR DESCRIPTION
**Pull request by ffvietpham**
Implementation of activity report v2 requests and refactoring of the LambdaProxyService to the new structure.

### CODE REVIEW

File: src/util/EnvironmentManagement.ts [hunk 1]
```diff
@@ -74,6 +74,19 @@ export class EnvironmentManagement {
...
+     * We only have DEV and PROD environment for activity report.
+     */
+    getActivityReportUrl(): string {
+        const stage = this.getStage();
+        if (stage === StageTypes.DEVELOPMENT) {
+            return 'https://latest-development-activity-report-v2-cloud.fe.flowfact-dev.cloud';
+        } else {
+            return 'https://latest-development-activity-report-v2-cloud.fe.flowfact-dev.cloud';
+        }
+    }
```
Comment (ffjanhoeck) [inline src/util/EnvironmentManagement.ts:88]: This is too specific and should not be in this file here

**Review by ffjanhoeck — CHANGES_REQUESTED** [review]
Because that is a breaking change, you should increase the version in the package.json

File: src/util/EnvironmentManagement.ts [hunk 2]
```diff
@@ -74,6 +74,19 @@ export class EnvironmentManagement {
...
+    /**
+     * URL for activity report based on stage
+     * We only have DEV and PROD environment for activity report.
+     */
+    getActivityReportUrl(): string {
+        const stage = this.getStage();
+        if (stage === StageTypes.DEVELOPMENT) {
+            return 'https://latest-development-activity-report-v2-cloud.fe.flowfact-dev.cloud';
+        } else {
+            return 'https://latest-development-activity-report-v2-cloud.fe.flowfact-dev.cloud';
```
Comment (ffvietpham) [inline src/util/EnvironmentManagement.ts:86]: This will be changed later, after we get the url for production configured

**Comment by ffjanhoeck** [comment]
Please also prettify your code!

File: src/util/EnvironmentManagement.ts (hunk 2 above)
Comment (rhoffmann) [reply src/util/EnvironmentManagement.ts:86]: in API Mapping there is a LambdaAPIService class,, where you can use a `forceUrl` parameter for this case, i think

File: src/service/ActivityReportService/ActivityReportV2Controller.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    /**
+     * publish activity report data as context file in the preconfigured AWS S3 bucket
+     * @param entityId entity id of the activity report object
+     */
+    async publicActivityReportV2(entityId: string): Promise<ApiResponse<LambdaServiceResponse>> {
+        const body = await this.prepareActivityReportV2Body('publish', entityId);
+        return this.invokeActivityReportV2(body);
+    }
+
+    async prepareEmailBody(links: ActivityReportLinkType[], message: string): Promise<ApiResponse<LambdaServiceResponse>> {
```
Comment (fftobiashoffmann) [inline src/service/ActivityReportService/ActivityReportV2Controller.ts:28]: can you add comments above

File: src/service/ImporterRecordLogService/ImporterRecordLogController.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { APIClient, APIMapping } from '../../http';
+
+export class ImporterRecordLogController extends APIClient {
+    constructor() {
+        super(APIMapping.lambdaProxyService);
+    }
+
+    async fetchLogDirectories() {
+        return this.invokeApiWithErrorHandling('/ff-importer-v2-record-log-s3-directory-entries', 'GET');
```
Comment (fftobiashoffmann) [inline src/service/ImporterRecordLogService/ImporterRecordLogController.ts:9]: can you add the type here?

File: src/service/ImporterRecordLogService/ImporterRecordLogController.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    }
+
+    async fetchDirectoryContents(table: string) {
+        return this.invokeApiWithErrorHandling('/ff-importer-v2-record-log-s3-directory-content', 'POST', { table });
```
Comment (fftobiashoffmann) [inline src/service/ImporterRecordLogService/ImporterRecordLogController.ts:13]: and here

File: src/service/ActivityReportService/ActivityReportV2Controller.ts (hunk 3 above)
Comment (ffvietpham) [reply src/service/ActivityReportService/ActivityReportV2Controller.ts:33]: @fftobiashoffmann , is done

File: src/service/ImporterRecordLogService/ImporterRecordLogController.ts (hunk 4 above)
Comment (ffvietpham) [reply src/service/ImporterRecordLogService/ImporterRecordLogController.ts:9]: Hi @fftobiashoffmann , we have just migrated this service to the new structure and don't maintain it. So we don't know about the type of the response

File: src/service/ImporterRecordLogService/ImporterRecordLogController.ts (hunk 5 above)
Comment (ffvietpham) [reply src/service/ImporterRecordLogService/ImporterRecordLogController.ts:13]: Hi @fftobiashoffmann , we have just migrated this service to the new structure and don't maintain it. So we don't know about the type of the response

**Review by fftobiashoffmann — APPROVED** [review]

**Pull request merged**